pkgs/dart_services/lib/src/generative_ai.dart

```diff
@@ -15,49 +14,71 @@ import 'pub.dart';
 final DartPadLogger _logger = DartPadLogger('gen-ai');
 
 class GenerativeAI {
-  // Valid values here are 'gemini-2.0-flash' and 'gemini-2.5-flash'.
-  static const String _geminiModel = 'gemini-2.5-flash';
+  // Valid values here are 'models/gemini-2.0-flash' and
+  // 'models/gemini-2.5-flash'.
+  static const String _geminiModel = 'models/gemini-2.5-flash';
   static const String _apiKeyVarName = 'GEMINI_API_KEY';
 
-  late final String? _geminiApiKey;
+  GenerativeService? gemini;
 
   GenerativeAI() {
-    final geminiApiKey = Platform.environment[_apiKeyVarName];
-    if (geminiApiKey == null || geminiApiKey.isEmpty) {
-      _logger.warning('$_apiKeyVarName not set; gen-ai features DISABLED');
-    } else {
+    try {
+      gemini = GenerativeService.fromApiKey();
+
       _logger.info('$_apiKeyVarName set; gen-ai features ENABLED');
-      _geminiApiKey = geminiApiKey;
+      // ignore: avoid_catching_errors
```

Comment:
@brianquinlan - `fromApiKey()` throws an ArgumentError if the key is not present in the environment. Should we instead throw ArgumentException?